tei_adapter/tei_id_extractor/src/main/scala/weco/catalogue/tei/id_extractor/GitHubBlobReader.scala

```diff
@@ -0,0 +1,71 @@
+package weco.catalogue.tei.id_extractor
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
+import uk.ac.wellcome.json.JsonUtil._
+
+import java.net.URI
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+
+class GitHubBlobReader(token: String)(implicit ac: ActorSystem) {
+  implicit val ec = ac.dispatcher
+  def getBlob(uri: URI): Future[String] = {
+    val request = HttpRequest(
+      uri = Uri(uri.toString),
+      headers = List(
+        Accept(
+          MediaType.applicationWithFixedCharset(
+            "vnd.github.v3+json",
+            HttpCharsets.`UTF-8`)),
+        `Cache-Control`(CacheDirectives.`no-cache`),
+        Connection("keep-alive"),
+        Authorization(OAuth2BearerToken(token))
+      )
+    )
+    for {
+      response <- Http().singleRequest(request)
+      blob <- unmarshalAs[Blob](response, request)
+      decoded <- Future.fromTry(decodeBase64(blob.content))
+    } yield decoded
+
+  }
+  private val base64 = java.util.Base64.getMimeDecoder
+
+  private def decodeBase64(s: String) = Try {
+    val bytes =
+      base64.decode(s.getBytes(java.nio.charset.StandardCharsets.UTF_8))
+    new String(bytes, java.nio.charset.StandardCharsets.UTF_8)
+  }
+
+  private def unmarshalAs[T](response: HttpResponse, request: HttpRequest)(
+    implicit um: Unmarshaller[ResponseEntity, T]): Future[T] = {
```

Comment:
why are the other ones different? maybe I'm not understanding how it works, but I though the circe support was just adding marshallers, not encodes/decoders, and if you have in scope your own decoder/encoders it will use those, or so I though

Reply:
Unless there's a strong opposition I think I'm going to keep the marshalling library. It still allows to pass an implicit decoder instead of an implicit unmarshaller, but it can create the unmarshaller from the decoder implicitly. Also, I like the control with the level of error details.
I'm going to change the method to have an implicit decoder instead of an implicit unmarshaller though